### ISSUE TITLE
[a11y]: NumberInput does not use aria-describedby to link help text with component - back-port to v10

### ISSUE THREAD

**Issue by nishanth-pothula** [carbon-design-system/carbon#14407]
### Package

@carbon/react

### Browser

Chrome

### Package version

v10.59.5

### React version

v16.2.0

### Description

Any helper text should be automatically surfaced to assistive technologies through the use of aria-describedby.
This should happen on ALL Carbon components that can have helper text. We are expecting it to be added to `NumberInput` and `TextArea` components.

This issue is resolved in carbon v11 in this PR https://github.com/carbon-design-system/carbon/pull/13371 and we are expecting it to be back-ported to carbon v10

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-numberinput--overview

### Steps to reproduce

Turn on JAWS and tab to NumberInput. JAWS will read the input, but will not announce the helper text.

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
This fix is included in [v.10.59.11](https://github.com/carbon-design-system/carbon/releases/tag/v10.59.11)
